code/campaignclassic/src/main/java/com/adobe/marketing/mobile/AEPPushTemplate.java

```diff
@@ -0,0 +1,409 @@
+/*
+  Copyright 2023 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+ */
+package com.adobe.marketing.mobile;
+
+import android.app.Notification;
+import android.app.NotificationManager;
+import android.os.Build;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.RequiresApi;
+
+import com.adobe.marketing.mobile.services.Log;
+import com.adobe.marketing.mobile.util.DataReader;
+import com.adobe.marketing.mobile.util.DataReaderException;
+import com.adobe.marketing.mobile.util.StringUtils;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class AEPPushTemplate {
```

Comment:
Yes, returning when an exception occurs will result in the instance creation and hence the members cannot be `final`. But, if you take the scenario here - absence of the required values imply that the payload is invalid and the creating the template simply is unusable. So IMO the right thing to do here (as I mentioned in a comment later) is to throw an `IllegalArgumentException` when you get a `DataReaderException`. This will also prevent the instance from being created and allow you to declare required members as final. Additionally such a pattern makes the caller responsible for handling what to do with a bad payload.